skglm/cv.py

```diff
@@ -0,0 +1,181 @@
+    """Cross-validated wrapper for GeneralizedLinearEstimator.
+
+    This class performs cross-validated selection of the regularization parameter(s)
+    for a generalized linear estimator, supporting both L1 and elastic-net penalties.
+
+    Parameters
+    ----------
+    datafit : object
+        Datafit (loss) function instance (e.g., Logistic, Quadratic).
+    penalty : object
+        Penalty instance with an 'alpha' parameter (and optionally 'l1_ratio').
+    solver : object
+        Solver instance to use for optimization.
+    alphas : array-like of shape (n_alphas,), optional
+        List of alpha values to try. If None, they are set automatically.
+    l1_ratio : float or array-like, optional
+        The ElasticNet mixing parameter(s), with 0 <= l1_ratio <= 1.
+        Only used if the penalty supports 'l1_ratio'. If None, defaults to 1.0 (Lasso).
+    cv : int, default=4
+        Number of cross-validation folds.
+    n_jobs : int, default=1
+        Number of jobs to run in parallel for cross-validation.
+    random_state : int or None, default=None
+        Random seed for cross-validation splitting.
+    eps : float, default=1e-3
+        Ratio of minimum to maximum alpha if alphas are set automatically.
+    n_alphas : int, default=100
+        Number of alphas along the regularization path if alphas are set automatically.
+
+    Attributes
+    ----------
+    alpha_ : float
+        Best alpha found by cross-validation.
+    l1_ratio_ : float or None
+        Best l1_ratio found by cross-validation (if applicable).
+    best_estimator_ : GeneralizedLinearEstimator
+        Estimator fitted on the full data with the best parameters.
+    coef_ : ndarray
+        Coefficients of the fitted model.
+    intercept_ : float or ndarray
+        Intercept of the fitted model.
+    alphas_ : ndarray
+        Array of alphas used in the search.
+    scores_path_ : ndarray
+        Cross-validation scores for each parameter combination.
+    n_iter_ : int or None
+        Number of iterations run by the solver (if available).
+    n_features_in_ : int or None
+        Number of features seen during fit.
+    feature_names_in_ : ndarray or None
+        Names of features seen during fit.
+    """
+
+    def __init__(self, datafit, penalty, solver, alphas=None, l1_ratio=None,
+                 cv=4, n_jobs=1, random_state=None,
+                 eps=1e-3, n_alphas=100):
+        super().__init__(datafit=datafit, penalty=penalty, solver=solver)
+        self.alphas = alphas
+        self.l1_ratio = l1_ratio
+        self.cv = cv
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+        self.eps = eps
+        self.n_alphas = n_alphas
+
+    def _score(self, y_true, y_pred):
+        """Compute the performance score (higher is better)."""
+        if isinstance(self.datafit, (Logistic, QuadraticSVC)):
+            return float(np.mean(y_true == y_pred))
+        return -float(np.mean((y_true - y_pred) ** 2))
```

Comment:
I'm +1 with using sklearn `accuracy_score` and `mean_squared_error`
this forces us to stay consistent with the sklearn API, as we are drop in replacement of it